assessments/projects/rook/self-assessment.md

```diff
@@ -0,0 +1,468 @@
+
+#### Workflows
+
+##### Block Storage Example
+
+In the diagram [here](#actors), the flow to create an application with an RWO volume is:
+
+1. The (blue) app creates a PVC to request storage
+2. The PVC defines the Ceph RBD storage class (sc) for provisioning the storage
+3. K8s calls the Ceph-CSI RBD provisioner to create the Ceph RBD image.
+4. The kubelet calls the CSI RBD volume plugin to mount the volume in the app
+5. The volume is now available for reads and writes.
+
+A ReadWriteOnce volume can be mounted on one node at a time.
+
+##### Shared File System Example
+
+In the diagram [here](#actors), the flow to create a applications with a RWX volume is:
+
+1. The (purple) app creates a PVC to request storage
+2. The PVC defines the CephFS storage class (sc) for provisioning the storage
+3. K8s calls the Ceph-CSI CephFS provisioner to create the CephFS subvolume
+4. The kubelet calls the CSI CephFS volume plugin to mount the volume in the app
+5. The volume is now available for reads and writes.
+
+A ReadWriteMany volume can be mounted on multiple nodes for your application to use.
+
+##### Block Storage Example
+
+In the diagram [here](#actors), the flow to create an application with access to an S3 bucket is:
+
+- The (orange) app creates an ObjectBucketClaim (OBC) to request a bucket
+- The Rook operator creates a Ceph RGW bucket (via the lib-bucket-provisioner)
+- The Rook operator creates a secret with the credentials for accessing the bucket and a configmap with bucket information
+- The app retrieves the credentials from the secret
+- The app can now read and write to the bucket with an S3 client
+
+A S3 compatible client can use the S3 bucket right away using the credentials (`Secret`) and bucket info (`ConfigMap`).
+
+
+### Goals
+
+####
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+#### Security Goals
+
+* All access to rook operator should be authenticated and authorized
+* Secrets created by Rook operator should maintain confidentiality
+* Rook operator should obey the principle of least privilege (see [here](https://github.com/rook/rook/pull/1736) for improvements already made to operator permissions model)
+* Ceph storage elements should maintain integrity and availability while scaling
+
+### Non-goals
+
+**General**
+
+* Rook is not meant to be used for storage outside of K8s
+* Rook is not meant to modify the Ceph or K8s architectures, but to leverage their existing capabilities in order to provide cloud native persistent storage
+
+**Security**
+
+* Ceph- and K8s-specific vulnerabilities
+* Vulnerabilities caused by user error
+* Address security issues of extensions or tools used with Rook
+
+## Self-assessment use
+
+This self-assessment is created by the Rook team to perform an internal analysis of the
```

Comment:
Was this created with active participation and review from the Rook team? Consider listing the authors of this self-assessment in the first lines, below the title header.

Reply:
Added authors and Rook level of participation [here](https://github.com/dhauss/tag-security/commit/f2318b659eac315a4b1e04001f2702eb467384e1). We had feedback from a maintainer, Travis Nielsen, but the Rook team was busy with a version update so his ability to participate was a bit limited

Reply:
Thanks @dhauss— Please adjust the phrasing to not state that this was created by the Rook team. The sentence may simply be omitted as you have the authorship documented at the top.

Reply:
Sounds good- I omitted the sentence, and also removed a reference to seeking graduation given Rook already graduated. You can find the changes in this [commit](https://github.com/dhauss/tag-security/commit/b89d1c4ac7670abb197293d7b8ef086d8ac63aef), happy to adjust further if needed